File: cookmate-front/cookmate-app/app/type/board.ts
export interface Board {
  boardNo: number;
  userNo: number;
  typeNo: number;
  boardTitle: string;
  introduce: string;
  imageUrl: string;
  url: string;
  open: string;
  likesCount: number;
  nickname: string;
  profileImageUrl: string;
  typeName: string;
  difficult: string;
  cookTime: string;
  calory: string;
  ai: string;
  isApiData?: string;
  recipeCount?: number;
  followerCount?: number;
  followingCount?: number;
  ingredientSets: IngredientSet[];
  cookSteps: CookStep[];

  boardPostdate: string;
}

export interface IngredientSet {
  setNo: number;
  setName: string;
  ingredients: Ingredient[];
}

export interface Ingredient {
  ingredientNo: number;
  ingredientName: string;
  quantity: string;
  unit: string;
}

export interface CookStep {
  step: number;
  cookContent: string;
  cookImage: string;
}

export interface BoardCard {
  boardNo: number;
  boardTitle: string;
  imageUrl: string;
  nickname: string;
  typeName: string;
  cookTime: string;
  ai: string;
  likesCount: number;
  scrapped: boolean;
}

export interface Comment {
  commentNo: number;
  parentCommentNo: number | null;
  userNo: number;
  nickname: string;
  profileImageUrl: string;
  commentContent: string;
  commentPostdate: string;
  replies: Comment[];
}

File: cookmate/src/main/java/com/kh/cookmate/board/model/vo/Board.java
package com.kh.cookmate.board.model.vo;

import java.util.Date;

import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
public class Board {
	private int boardNo;
    private int userNo;
    private int typeNo;
    private String boardTitle;
    private String nickname;
    private String introduce;
    private Date boardPostdate;
    private String imageUrl;
    private String url;           // 유튜브 URL
    private int likesCount;
    private char boardDelete;     // N: 정상, Y: 삭제
    private char open;            // N: 비공개, Y: 공개
	private char isApiData;
	private String apiRecipeId;
	
}
